iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/KubernetesSensor.java

```diff
@@ -80,14 +86,13 @@ private static boolean hasKubernetesObjectStructure(InputFile inputFile) {
         String[] lines = LINE_TERMINATOR.split(text);
         boolean hasExpectedIdentifier = false;
         for (String line : lines) {
-          if (containsHelmChartTemplateDirective(line)) {
-            LOG.debug("Line contains help chart directive, file will not be analyzed.\n{}", line);
-            return false;
-          }
           if (IDENTIFIER.stream().anyMatch(line::startsWith)) {
             identifierCount++;
           } else if (FILE_SEPERATOR.equals(line)) {
             identifierCount = 0;
+          } else if (line.contains("{{") && !HELM_DIRECTIVE_IN_COMMENT_OR_STRING.matcher(line).find()) {
+            // Do not parse Helm files with template directives
+            return false;
```

Comment:
```suggestion
            LOG.debug("Line contains help chart directive, file will not be analyzed.\n{}", line);
            return false;
```
Maybe this log can be useful.